services/eventservice/src/tests.rs

```diff
@@ -0,0 +1,508 @@
+use crate::types::Category;
+
+use super::*;
+
+#[test]
+fn merge_test() {
+    assert_eq!(
+        data_manager::EventManager::_merge(b"", Some(b"ab"), b"cd"),
+        Some((b"abcd").to_vec())
+    );
+}
+
+#[test]
+fn compare_by_val_test() {
+    let event1 = types::Event {
```

Comment:
please create a separate mock factory or implement the function that will return a certain amount of objects in loop instead of manually creating lots of objects again and again in test file.

Reply:
Done